src/terminal.c

```diff
@@ -5543,8 +5915,61 @@ term_and_job_init(
     garray_T	    ga_cmd, ga_env;
     char_u	    *cmd = NULL;
 
-    if (dyn_winpty_init(TRUE) == FAIL)
-	return FAIL;
+    has_winpty = dyn_winpty_init(FALSE) != FAIL ? TRUE : FALSE;
+    has_conpty = dyn_conpty_init(FALSE) != FAIL ? TRUE : FALSE;
+
+    if (!has_winpty && !has_conpty)
+	return dyn_winpty_init(TRUE);
+
+    if (opt->jo_term_mode == 'w')
+	set_string_option_direct((char_u *)"tmod", -1, (char_u *)"winpty",
+							OPT_FREE|OPT_LOCAL, 0);
+    if (opt->jo_term_mode == 'c')
+	set_string_option_direct((char_u *)"tmod", -1, (char_u *)"conpty",
+							OPT_FREE|OPT_LOCAL, 0);
+
+    if (curwin->w_p_tmod == NULL || *curwin->w_p_tmod == NUL)
```

Comment:
You can refactor around here.
I think that it is not preferable to use `goto` sentences in parts that are not error handling.

Reply:
That is your idea. My idea is different.
I do not know which is right. But this will work.
I use tools. That's all.

Reply:
For example, the L5934 - L5937 can simply change to one line of `goto to_conpty;`.
Do you refuse such a thing?

The fact that it is working is not the most important.
You are thinking too much of just the facts that are working.
Consider the maintainability of the code.
Vim's main maintainer is Bram.  not you.

Reply:
Do not get involved. Do not push it. the end.

Reply:
I have not talked about code style.
I am suggesting making code easier to read and making it simple.

@brammool >
Include @ntak's patch.
I will write a patch after that.
Perhaps someone will write even if I do not write.